sktime/classification/feature_based/_adi_cv_classifier.py

```diff
@@ -0,0 +1,198 @@
+        ----------
+        adi_threshold : float, optional
+            Threshold for Average Demand Interval. Defaults to 1.32.
+
+        cv_threshold : float, optional
+            Threshold for Variance. Defaults to 0.49.
+
+        features : List[str] | None, optional
+            List of features to compute. Defaults to None (all features)
+
+        Raises
+        ------
+            ValueError: If features is provided and does not
+            contain 'adi','cv2', or 'class'.
+        """
+        self.adi_threshold = adi_threshold
+        self.cv_threshold = cv_threshold
+        self.features = features
+
+        # Initialize the parent class
+        super().__init__()
+
+        # Checking if the features parameter is valid
+        if features is not None:
+            if (
+                "adi" not in features
+                or "cv2" not in features
+                or "class" not in features
+            ):
+                raise ValueError(
+                    "The features list must either be None or include adi "
+                    + "cv2, and class as elements."
+                )
+
+        else:
+            # Helpful to transform None to default list for transform function
+            self.features = ["adi", "cv2", "class"]
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series
+            Series of time series data this transform function aims to classify
+
+        y : Series | None, default=None
+            Not required for the classification of the time series
+
+        Returns
+        -------
+        X_transformed : pd.DataFrame
+            Contains columns listed in self.features and 1 row entry of values
+
+        """
+        condition = X != 0
+        X_non_zero = X[condition]
+
+        # Calculating ADI value based on formula from paper
+        adi_value = len(X) - 1 / (len(X_non_zero) - 1)
+
+        # Calculating variance for all non-zero values
+        variance = X_non_zero.var()
+        cv2_value = variance / len(X_non_zero)
+
+        # Calculating the class type
+        class_type = ""
+
+        if adi_value <= self.adi_threshold:
+            if cv2_value <= self.cv_threshold:
+                class_type = "smooth"
+
+            else:
+                class_type = "erratic"
+
+        elif cv2_value <= self.cv_threshold:
+            class_type = "intermittent"
+
+        else:
+            class_type = "lumpy"
+
+        # Collecting all values together into dict and converting to DF
+        return_dict = {}
+
+        if "adi" in self.features:
+            return_dict["adi"] = adi_value
+
+        if "cv2" in self.features:
+            return_dict["cv2"] = cv2_value
+
+        if "class" in self.features:
+            return_dict["class"] = class_type
+
+        df = pd.DataFrame(return_dict)
```

Comment:
for consistent user erxperience, here I would reorder the columns in the same order as passed in `feature`.

Reply:
Got it @fkiraly, I'll run through the features in a for-loop and add them sequentially in that manner! 😄

Reply:
A simpler way might be just to do `my_df.loc[:, my_ordered_list]` on the final object before returning?

Reply:
Got it, I'll write the code to do it like that